File: words-api/Lib/BridgeTypes/PronounKind.cs
// Project: words-api
// File: PronounKind.cs
// 
// Author: Ethan Kisiel (ethan.a.kisiel@gmail.com)
// 
// File Created: 23 08 2025 22:08:44
// Last Modified: 23 08 2025 23:08:20
// 
// Modified By: Ethan Kisiel (ethan.a.kisiel@gmail.com>)
// 
// Copyright 2025 - 2025 Ethan Kisiel, Ethan Kisiel

namespace words_api.Lib.Enums;

public class PronounKind
{
    public const string Default = "X";
    public const string Personal = "PERS";
    public const string Relative = "REL";
    public const string Reflexive = "REFLEX";
    public const string Demonstrative = "DEMONS";
    public const string Interrogative = "INTERR";
    public const string Indefinite = "INDEF";
    public const string Adjectival = "ADJECT";

    // X,      --  unknown, nondescript
    // Pers,   --  PERSonal
    // Rel,    --  RELative
    // Reflex, --  REFLEXive
    // Demons, --  DEMONStrative
    // Interr, --  INTERRogative
    // Indef,  --  INDEFinite
    // Adject  --  ADJECTival
}

File: words-frontend/src/lib/models/PronounKind.ts
export type PronounKindValue =
    'X' | 'PERS' | 'REL' | 'REFLEX' | 'DEMONS' | 'INTERR' | 'INDEF' | 'ADJECT';

export class PronounKind {

    private static default = 'X';
    private static personal = 'PERS';
    private static relative = 'REL';
    private static reflexive = 'REFLEX';
    private static demonstrative = 'DEMONS';
    private static interrogative = 'INTERR';
    private static indefinite = 'INDEF';
    private static adjectival = 'ADJECT';

    static getLongForm(value: PronounKindValue, detailed=false): string {
        switch (value) {
            case this.default:
                return (!detailed?
                    'Unknown' :
                    'Default/unspecified pronoun kind');
            case this.personal:
                return (!detailed?
                    'Personal' :
                    'Personal pronoun');
            case this.relative:
                return (!detailed?
                    'Relative' :
                    'Relative pronoun');
            case this.reflexive:
                return (!detailed?
                    'Reflexive' :
                    'Reflexive pronoun');
            case this.demonstrative:
                return (!detailed?
                    'Demonstrative' :
                    'Demonstrative pronoun');
            case this.interrogative:
                return (!detailed?
                    'Interrogative' :
                    'Interrogative pronoun');
            case this.indefinite:
                return (!detailed?
                    'Indefinite' :
                    'Indefinite pronoun');
            case this.adjectival:
                return (!detailed?
                    'Adjectival' :
                    'Adjectival pronoun');
            default:
                return '';
        }
    }
}